### PR TITLE
Fixing today helper when scrolled below the fold...

### LINKED ISSUE #24: Floating Today Helper not reliable
Ive noticed some bugs with the floating today helper. My interpretation of the helper is that if "today" is not visible on the screen, then the helper appears.

1) If you scroll all the way to the top, it hides. I think it should stay visible.
2) When scrolling back in time, the helper does not appear until "today" is several weeks below the fold.
3) When scrolling forwards in time, the helper does not appear until "today" is several months above the fold.

I think it would be more convenient if the today helper was always visible if "today" was not visible.

For these screenshots, today is June 24th.

screen shot 2016-06-24 at 11 57 28 am
screen shot 2016-06-24 at 11 59 54 am

### PR DESCRIPTION
**Pull request by offsky**
Also improving animation of chevron when transitioning from pointing up to hiding
Fixes #24

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -237,9 +237,9 @@ export default class InfiniteCalendar extends Component {
 		let newState;
 
 		if (scrollTop >= this.todayOffset + height - monthHeight + rowHeight * todayHelperRowOffset) {
-			if (showToday !== 1) newState = 1;
-		} else if (scrollTop <= this.todayOffset - rowHeight * todayHelperRowOffset) {
-			if (showToday !== -1) newState = -1;
+			if (showToday !== 1) newState = 1; //today is above the fold
+		} else if (scrollTop <= this.todayOffset - monthHeight - rowHeight * todayHelperRowOffset) {
```
Comment (clauderic) [inline src/index.js]: I believe this is incorrect. `todayOffset` is the scrollTop offset for the month `today` appears in. If you subtract the monthHeight you get an incorrect result that's approximately around the begining of the scrollTop offset of the previous month.

The logic counts the number of rows before the start of the current month / the number of rows after the end of the current month.

To ensure the helper is not visible when today is visible, you only need to tweak the `todayHelperRowOffset` prop relative to the calendar's `height`.

Makes sense?

**Comment by clauderic** [comment]
Hey @offsky, thanks for submitting this. Happy to accept your PR with regards to the css fixes to the today arrow. See my comments regarding your changes to the offset logic though.

**Comment by offsky** [comment]
Is it possible to get the scrollTop offset of the day "today" instead of the month that "today" is inside of?  This would be the most accurate.  I can get close by doing `round(date/7)`. This will be more accurate, but may be off by 1 a few times a year.  

Because my implementation of your calendar has a flexible height, I would prefer to find a general purpose solution instead of having to tune todayHelperRowOffset based on the height.

**Comment by clauderic** [comment]
Sure, that's probably a better implementation actually! If you don't mind getting your hands dirty and updating your PR, the easiest way would probably be something along the lines of `todayOffset = monthOffset + todayEl.offsetTop`, just because `round(date/7)` is unreliable

**Comment by offsky** [comment]
Ok, here is my update. I went with a refined version of date/7 which takes the edge cases into account. I tested it with a number of months and "today"s that I knew were edge cases and it worked correctly each time.  With this commit, the "todayHeplerRowOffset" prop means the number of rows that "today" is scrolled away.  And its the same in either direction.  As a bonus, we no longer need to know the height of the month so I removed that calculation.

**Comment by clauderic** [comment]
Hey @offsky, thanks for updating this. I'll take some time to review it later today, but at first sight it looks sound 👍

**Comment by clauderic** [comment]
Works great, I'll go ahead and merge this! Thanks for all your work on this 👍 Nice touch on the arrow css transition, I dig it :)

**Pull request merged**